cci_nepal/analysis/dummy_data/dummy_data_modeling/model_analysis.py

```diff
@@ -0,0 +1,295 @@
+X = pd.get_dummies(data=X, drop_first=True)
+logging.info(X.columns)
+X
+
+
+nfri_basic = [
+    "Plastic Tarpaulin",
+    "Blanket",
+    "Sari",
+    "Male Dhoti",
+    "Shouting Cloth / Jeans",
+    "Printed Cloth",
+    "Terry Cloth",
+    "Utensil Set",
+    "Water Bucket",
+    "Nylon Rope",
+    "Sack (Packing Bag)",
+]
+
+nfri_non_basic = [
+    "Cotton Towel",
+    "Laundry Soap",
+    "Tooth Brush and Paste",
+    "Sanitary Pad",
+    "Ladies Underwear",
+    "Torch Light",
+    "Whistle Blow",
+    "Nail Cutter",
+    "Hand Sanitizer",
+    "Liquid Chlorine",
+]
+
+y_basic = df_dummy.loc[:, nfri_basic]
+y_basic
+
+y_non_basic = df_dummy.loc[:, nfri_non_basic]
+y_non_basic
+
+X_train_basic, X_test_basic, y_train_basic, y_test_basic = train_test_split(X, y_basic)
+(
+    X_train_non_basic,
+    X_test_non_basic,
+    y_train_non_basic,
+    y_test_non_basic,
+) = train_test_split(X, y_non_basic)
+
+
+model_basic.fit(X_train_basic, y_train_basic)
+model_non_basic.fit(X_train_non_basic, y_train_non_basic)
+
+
+test_predictions_basic = model_basic.predict(X_test_basic)
+test_predictions_non_basic = model_non_basic.predict(X_test_basic)
+
+test_predictions_labels_basic = [
+    [numeric_score_transformer(i) for i in nested] for nested in test_predictions_basic
+]
+test_predictions_items_wise_basic = list(
+    map(list, zip(*test_predictions_labels_basic))
+)  # Transforming the above list
+
+test_predictions_labels_non_basic = [
+    [numeric_score_transformer(i) for i in nested]
+    for nested in test_predictions_non_basic
+]
+test_predictions_items_wise_non_basic = list(
+    map(list, zip(*test_predictions_labels_non_basic))
+)  # Transforming the above list
+
+
+accuracy_list_basic = []
+for i in range(0, len(nfri_basic)):
+    accuracy_list_basic.append(
+        accuracy_score(test_predictions_items_wise_basic[i], y_test_basic.iloc[:, i])
+    )
+
+logging.info(
+    pd.DataFrame(accuracy_list_basic, y_test_basic.columns, columns=["Accuracy"])
+)
+
+
+accuracy_list_non_basic = []
+for i in range(0, len(nfri_non_basic)):
+    accuracy_list_non_basic.append(
+        accuracy_score(
+            test_predictions_items_wise_non_basic[i], y_test_non_basic.iloc[:, i]
+        )
+    )
+
+logging.info(
+    pd.DataFrame(
+        accuracy_list_non_basic, y_test_non_basic.columns, columns=["Accuracy"]
+    )
+)
+
+
+model_basic.fit(X, y_basic)
+model_non_basic.fit(X, y_non_basic)
+
+# Making prediction
+
+row_basic = X.iloc[
+    10,
+].tolist()
+yhat_basic = model_basic.predict([row_basic])
+
+# Summarizing prediction
+
+logging.info("Printing the scores for NFRI Basic")
+logging.info(yhat_basic[0])
+
+
+logging.info("Now printing the scores for NFRI Non Basic")
+
+
+# Making prediction
+
+row_non_basic = X.iloc[
+    10,
+].tolist()
+yhat_non_basic = model_non_basic.predict([row_non_basic])
+
+# Summarizing prediction
+
+logging.info(yhat_non_basic[0])
+
+
+# Plotting the Coefficient scores of the Model
```

Comment:
The below is a little confusing to interpret. Maybe a bar chart with the feature names on the x axis and the scores on the Y axis would work best?